iot_verifier/src/packet_loader.rs

```diff
@@ -17,52 +28,113 @@ pub enum NewLoaderError {
 }
 
 impl PacketLoader {
+    pub fn from_settings(settings: &Settings, pool: PgPool) -> Self {
+        tracing::info!("from_settings packet loader");
+        let cache = settings.cache.clone();
+        Self { pool, cache }
+    }
+
     pub async fn run(
-        &mut self,
+        &self,
         mut receiver: Receiver<FileInfoStream<IotValidPacket>>,
         shutdown: &triggered::Listener,
         gateway_cache: &GatewayCache,
+        file_upload_tx: FileUploadSender,
     ) -> anyhow::Result<()> {
         tracing::info!("starting verifier iot packet loader");
+        let store_base_path = Path::new(&self.cache);
+        let (non_rewardable_packet_sink, mut non_rewardable_packet_server) =
+            file_sink::FileSinkBuilder::new(
+                FileType::NonRewardablePacket,
+                store_base_path,
+                concat!(env!("CARGO_PKG_NAME"), "_non_rewardable_packet"),
+                shutdown.clone(),
+            )
+            .deposits(Some(file_upload_tx.clone()))
+            .roll_time(ChronoDuration::minutes(5))
+            .create()
+            .await?;
+        tokio::spawn(async move { non_rewardable_packet_server.run().await });
```

Comment:
```suggestion
        tokio::spawn(non_rewardable_packet_server.run());
```
i believe you can remove the async block when you're only doing a single awaitable async operation within the spawned task